crates/subspace-service/src/sync_from_dsn/fast_sync.rs

```diff
@@ -260,6 +263,7 @@ where
         );
 
         let mut import_block = BlockImportParams::new(
+            // We support the case when the block with extra-big blocks.
```

Comment:
Hm... this comment is in various places but I don't think it actually explains anything.

Here specifically the reason we have this check is that the block we are importing is the last one, there are no other blocks to import afterwards, hence we need to notify Substrate sync with `BlockOrigin::NetworkBroadcast` instead of `BlockOrigin::NetworkInitialSync`.

I don't see how `We support the case when the block with extra-big blocks.` would help anyone understand that, I don't see a direct connection at all.